Jenkinsfile

```diff
@@ -0,0 +1,30 @@
+#!groovy
+
+node {
+  checkout scm
+
+  def dockerRepoName = 'zooniverse/annotate'
+  def dockerImageName = "${dockerRepoName}:${env.BUILD_ID}"
+  def newImage = null
+
+  stage('Build Docker image') {
+    newImage = docker.build(dockerImageName)
+    newImage.inside {
+      sh 'npm install'
+    }
+  }
+
+  stage('Deploy') {
+    if (BRANCH_NAME == 'staging') {
```

Comment:
I've found on zooniverse/Panoptes-Front-End#4753 that Jenkins sets `BRANCH_NAME` to the name of the PR branch eg. `PR-4753`. If you figure out how to get the git branch name, I'd love to know for that PR too.

Reply:
If it's a push to a branch it'll be the branch name (unless that branch is a PR). If it's a merge event to a target, e.g. merge PR to master it'll be 'master'. 

PR's will be named like you mention and any PR made from a branch will override the previous branch builds, e.g. dev, comit, push branch will create an event in Jenkins and any PR made from that branch will take over.

Reply:
Also just adding to this. In panoptes we auto deploy any master branch changes to the staging env for testing (PFE to production). We use a production tag in panoptes to signal a production deploy and we update that tag via a lita cmd. 

Should we switch to something similar to avoid having a staging branch? I.e. autodeploy master to staging / test env and production branch / tag for production releases when we are happy?

Reply:
I'm not sure if using the same process on front end apps as the back end would make sense. With front end PRs, there's often some manual testing, especially if the changes have to do with CSS styles. Having a staged branch can make it easier to do testing on mobile or using a VM (whether its local in virtualbox or on browserstack).

Reply:
Right - my point was mostly that the `if (BRANCH_NAME == 'staging')` will never fire as there is no staging branch in this repo. Hence how you get the staging code deployed to a target domain for testing? 

Will it vary based on PR's (`preview.zooniverse.org/pr-123-annotate.org.uk`) or do you want a consistent up to date state that is ready `annotate.preview.zooniverse.org` or the like?

Reply:
ah, I see yeah. For CFEs like this, I don't think staged branches are as important as it's often only one or maybe two devs who work on them.  I would just do the latter at a domain like `annotate.preview.zooniverse.org`.